test/apps/iodemo/io_demo.cc

```diff
@@ -766,9 +826,24 @@ class DemoClient : public P2pDemoCommon {
             }
             std::cout << std::endl;
         }
+        if (opts().print_conn_status) {
+            std::cout << get_time_str() << " ";
+            std::cout << "io total ";
+            for (int i = 0; i < _conn.size(); i++) {
+                std::cout << _conn[i].num_completed << " ";
+            }
+            std::cout << std::endl;
+        }
     }
 
 private:
+    typedef struct {
+        long           num_sent;
+        long           num_completed;
+        UcxConnection* conn;
+    } conn_info_t; 
```

Comment:
pls move this to connection class then all the rest will be much simpler

Reply:
@evgeny-leksikov Maybe we can't do this since we need to keep the values of ``num_sent & num_completed`` for printing status per connection when the connection disconnect and reconnect again. But once the connection is disconnected then the UcxConnection is also deleted, the next time reconnect will new another different UcxConnection.

Reply:
@leibin2014 linear search of `conn/conn_info_t` (actually twice - started/completed) per operation will impact performance on scale. how about do drop values on destructor and maybe periodically if it's needed before disconnect?

Reply:
@evgeny-leksikov You are right it can impact performance on scale. But we should not drop the values when disconnect(on destructor) since we always want print out whole io counter on the connection with same server include previous connections' io counter. I did some more changes to optimize the performance issue as you mentioned. Please have one more look. Thanks!

Reply:
/azp run

Reply:
I'm not sure we want to print numbers for every connection every time, at scale it will be a lot of data in logs. IMO better to print aggregate performance of client and probably numbers of specific connection on disconnect as additional info.
@yosefe WDYT?

Reply:
maybe need to find a compact way to print it, but we do want to print status for all connections to make sure no connection is "dead". maybe print min/max/avg so if min==0 it means one connection is "dead"

Reply:
@yosefe @evgeny-leksikov 
I think we are discussing two things. Here just want to clarify what we really need.
1. Do we need to change the printing format? Current printing format ``[1250000.56567] io total 100 1200 900 ...`` was suggested by Yossi. I added option ``-x`` to enable and disable it, and print the status in ``report_performance()`` which will be called every 1 second. Do we want to change the format to ``[1250000.56567] io total min/max/avg`` to simplify the log?
2. Do we want to put the previous io counter before reconnect into io total? As Yossi's original suggestion, even if the connection is disconnect, we still need to print the io counter which is increased before disconnect and after reconnect the io total should be increased from the previous io counter number, but not from 0. So we should not drop the counter when disconnect. Do we need to change this?

Reply:
1. Yes, change to mix/max/avg. We can print it in the same line we print total bw today, so we can print something like this
`[%lu.%lu] %lu reads, %lu writes, total %.2f MB/s, %u conns min %.2f max %.2f avg %.2f"

2. if we print min/max/avg BW per (1) no need to save io counter, since (1) will give enough indication that all connections are ok

Reply:
maybe also print non-empty list of servers which don't respond in extra line?

Reply:
@yosefe About (2), then it can't indicate the io balance for all connections since even if a short time of disconnect/connect can clear the counter to 0. Do we just need the values to check the connection link status? Then why don't we just print a connect error log from dispatch_connection_error()?

Reply:
- it's ok that the minimum would be 0 for short period of time - as long as it becomes non-0 soon enough
- just printing connection error is not enough: we also want to make sure data is actually sent/received over every connection. this is a more strict check, than just looking for error

Reply:
@yosefe I got it.
@yosefe @evgeny-leksikov 
In current implementation we can get the total io of each connection without clear it when disconnect. I think this can show more information than the above solution.
It will just aviod to put the total io counters into UcxConnection. For sure put the io counters into UcxConnection can simplify some code in iodemo as @evgeny-leksikov mentioned. But another reason of not put them into UcxConnection is that the read/write io counters are not a common concept of UcxConnection, they are more like some concept of io demo. So I still prefer to keep the io counters in io demo, but not in  UcxConnection. Let me know please if you still think we need to change it, then I will change the current implementation.

Reply:
agree, we should keep the counters outside of UcxConnection because this is io_demo concept

Reply:
> maybe also print non-empty list of servers which don't respond in extra line?

@evgeny-leksikov If non-empty list is big at scale. We can be back to the trouble of too much log.

Reply:
> @evgeny-leksikov If non-empty list is big at scale. We can be back to the trouble of too much log.

ok

So, pls rename `conn_info_t` to `server_info_t` since it represents server's status, not a connection.
Also need to put server index to connection on client side for quick access.

Reply:
@evgeny-leksikov Fixed. Thanks!